projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -916,7 +935,15 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
      * @memberOf {@link IgxDatePickerComponent}
      */
     public selectDate(date: Date): void {
+        const oldDate = this.value;
         this.value = date;
+        if ((oldDate && oldDate.getTime()) !== (this.value && this.value.getTime())) {
```

Comment:
you have this same code in three locations. I would extract this to private method like:
`emitOnValueChanged(oldValue: Date, newValue: Date)`

Reply:
Or erm, the value set process could go through one place, rather than.. 4 it seems? 🙄